### PR TITLE
SMP-286 feat: flutter updated

### PR DESCRIPTION
**Pull request by mark-kascheev**
## Changes
Update flutter version

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assignee and assigned mandatory reviewers.
- [x] Required labels marked
- [x] Specified related tasks and/or related PRs.
- [x] Specified Changes.
- [ ] Attached videos/screenshots demonstrating the fix/feature.
- [ ] All unspecified fields in the PR description deleted.
- [ ] New code covered by tests.

## Checklist for reviewers
- [ ] CI passed successfully _(with a green check mark)_.
- [ ] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [ ] System design corresponds to the agreements on structure and architecture on the project.
- [ ] The code is decomposed into necessary and sufficient components.

Functionality:
- [ ] The code solves the problem.
- [ ] Any changes to the user interface are reasonable and look good.

Complexity:
- [ ] The code is clear, easy to read, functions are small, no more than 50 lines.
- [ ] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [x] The tests are correct, helpful, and well designed/developed.

Naming:
- [ ] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [ ] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

**Review by dshevchenkoo — APPROVED** [review]

File: example/.fvm/flutter_sdk [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/markascheev/fvm/versions/3.7.3
```
Comment (dshevchenkoo) [inline example/.fvm/flutter_sdk:1]: this file should be added to git ignore

File: .fvm/flutter_sdk [hunk 2]
```diff
@@ -0,0 +1 @@
+/Users/markascheev/fvm/versions/3.7.3
```
Comment (dshevchenkoo) [inline .fvm/flutter_sdk:1]: this file should be added to git ignore

Comment (mark-kascheev) [reply .fvm/flutter_sdk]: ✅

File: .fvm/flutter_sdk [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/parshonok/fvm/versions/3.7.6
```
Comment (dshevchenkoo) [inline .fvm/flutter_sdk:1]: add to ignore

File: example/.fvm/flutter_sdk [hunk 4]
```diff
@@ -0,0 +1 @@
+/Users/parshonok/fvm/versions/3.7.6
```
Comment (dshevchenkoo) [inline example/.fvm/flutter_sdk:1]: add to ignore

File: example/lib/main.dart [hunk 5]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// ignore_for_file: library_private_types_in_public_api
```
Comment (dshevchenkoo) [inline example/lib/main.dart:15]: ??

Comment (mark-kascheev) [reply example/lib/main.dart:15]: New lint rules warn in statefull widgets that is not okay define private class

File: .fvm/flutter_sdk (hunk 3 above)
Comment (mark-kascheev) [reply .fvm/flutter_sdk:1]: I fixed it in SNP-271

File: example/.fvm/flutter_sdk (hunk 4 above)
Comment (mark-kascheev) [reply example/.fvm/flutter_sdk:1]: I fixed it in SNP-271

**Review by ruinalsu — DISMISSED** [review]

**Comment by plasticfiresam** [comment]
Will not be merged, SDK constraints will be updated in separate PR

**Pull request closed without merging**